lib/Index/Index.cpp

```diff
@@ -153,6 +154,25 @@ class IndexSwiftASTWalker : public SourceEntityWalker {
 
   bool IsModuleFile = false;
   bool isSystemModule = false;
+
+  struct ContainerStackEntry {
+    const Decl *VisitedDecl;
+    // These vector capacities represent the number of declarations that may
+    // be associated with the key type. In most cases there will be only one; a
+    // function or single named pattern binding. However in the case of an
+    // expanded tuple pattern binding, there will be multiple.
+    SmallVector<const Decl *, 6> ActiveDecls{};
+    typedef const void *DeclMapKey;
+    llvm::DenseMap<DeclMapKey,
+                   llvm::PointerUnion<const Decl *, const Pattern *>>
+        DeclMap{};
+  };
+  SmallVector<ContainerStackEntry, 4> ContainerStack;
+  typedef llvm::Optional<
+      llvm::PointerUnion<const VarDecl *, const TuplePattern *>>
+      PatternElement;
+  typedef SmallVector<PatternElement, 6> PatternElements;
```

Comment:
I'm not sure we did. I just picked 6 since it seemed like a reasonable number to cover most TuplePatterns. Happy to tweak if you think more or less would be better.

Reply:
It'd be good to get some data on this, ie. by looking at tuple patterns in the source compatibility suite. 4 is probably plenty for most cases I'd think (?)

Reply:
Moot point now with the latest changes.